src/hevm/src/EVM/SymExec.hs

```diff
@@ -225,33 +266,58 @@ pruneDeadPaths =
     Just (VMFailure DeadPath) -> False
     _ -> True
 
+consistentPath :: VM -> Query (Maybe VM)
+consistentPath vm = do
+  resetAssertions
+  constrain $ sAnd $ fst <$> view constraints vm
+  checkSat >>= \case
+    Sat -> return $ Just vm
+    _   -> return $ Nothing
```

Comment:
do we maybe want to distinguish between unsat results and timeouts here?

Reply:
yes, good idea!